File: javascript/perfect-numbers/perfect-numbers.js
//Perfect number exercice
export const classify = (num) => {
  if(num <= 0) throw new Error('Classification is only possible for natural numbers.')
  let aux = 0;
  for(let i = 1; i < num; i++){
    if(num % i === 0){
      aux += i     
    }    
  }
  if(aux === num) return 'perfect';
  if(aux < num) return 'deficient'
  if(aux > num) return 'abundant'
};

File: python/perfect-numbers/perfect_numbers.py
def classify(number):
    if(number <= 0):
        raise ValueError('Error, number invalid')
    result = sum([i for i in range(1, number) if number % i == 0])
    if (number == result):
        return 'perfect'
    elif (number < result):
        return 'abundant'
    else:
        return 'deficient'
